src/hyperv/mod.rs

```diff
@@ -109,9 +109,31 @@ pub mod report {
     }
 
     fn hcl_report(bytes: &[u8]) -> Result<AttestationReport> {
-        let hcl: Hcl =
-            bincode::deserialize(bytes).context("unable to deserialize bytes from vTPM")?;
+        use std::mem::size_of;
+
+        struct HclLayout;
+        impl HclLayout {
+            const RSV1_SIZE: usize = size_of::<u32>() * 8;
+            const REPORT_SIZE: usize = 1184;
+            const RSV2_SIZE: usize = size_of::<u32>() * 5;
+            const TOTAL_SIZE: usize = Self::RSV1_SIZE + Self::REPORT_SIZE + Self::RSV2_SIZE;
+            const REPORT_RANGE: std::ops::Range<usize> =
+                Self::RSV1_SIZE..(Self::RSV1_SIZE + Self::REPORT_SIZE);
+        }
+
+        if bytes.len() < HclLayout::TOTAL_SIZE {
+            return Err(anyhow!(
+                "HCL report size mismatch: expected at least {}, got {}",
+                HclLayout::TOTAL_SIZE,
+                bytes.len()
+            ));
+        }
+
+        let report_bytes = bytes
+            .get(HclLayout::REPORT_RANGE.clone())
```

Comment:
By moving the constants outside, you won't need to use .clone() here
```suggestion
            .get(REPORT_RANGE)
```

Reply:
I'll do a final squash of all my commits before merge once I've addressed everyone's concerns.